moveit_planners/pilz_industrial_motion_planner/src/trajectory_generator_lin.cpp

```diff
@@ -134,9 +133,7 @@ void TrajectoryGeneratorLIN::extractMotionPlanInfo(const planning_scene::Plannin
     info.start_joint_position[joint_name] = req.start_state.joint_state.position[index];
   }
```

Comment:
```suggestion
for (unsigned i=0; i<req.start_state.joint_state.name.size(); ++i) {
    info.start_joint_position[req.start_state.joint_state.name[i]] = req.start_state.joint_state.position[i];
  }
  for (const auto& joint_name : robot_model_->getJointModelGroup(req.group_name)->getActiveJointModelNames())
  {
    if (info.start_joint_position.count(joint_name) == 0)
    {
      std::ostringstream os;
      os << "Could not find joint \"" << joint_name << "\" of group \"" << req.group_name
         << "\" in start state of request";
      throw LinJointMissingInStartState(os.str());
    }
  }
```

Reply:
^ (replacing everything inside that `{}` (I can't work out how to do multi-line suggestions))

looks like it correctly updates all the joint states, including those not in the current joint group.